nighthawk/source/common/statistic_impl.cc

```diff
@@ -0,0 +1,216 @@
+#include "nighthawk/source/common/statistic_impl.h"
+
+#include <cmath>
+#include <sstream>
+#include <stdio.h>
+
+#include "common/common/assert.h"
+
+namespace Nighthawk {
+
+std::string StatisticImpl::toString() const {
+  std::stringstream stream;
+  stream << fmt::format("#Completed: {}. Mean: {:.{}f}μs. Stdev: {:.{}f}μs.", count(),
+                        mean() / 1000, 2, stdev() / 1000, 2)
+         << std::endl;
+  return stream.str();
+}
+
+void StatisticImpl::toProtoOutput(nighthawk::client::Output& output) {
+  output.set_request_count(count());
+  output.mutable_mean()->set_nanos(mean());
+  output.mutable_stdev()->set_nanos(stdev());
+}
+
+StreamingStatistic::StreamingStatistic() : count_(0), mean_(0), sum_of_squares_(0) {}
+
+void StreamingStatistic::addValue(int64_t value) {
+  double delta, delta_n;
+  count_++;
+  delta = value - mean_;
+  delta_n = delta / count_;
+  mean_ += delta_n;
+  sum_of_squares_ += delta * delta_n * (count_ - 1);
```

Comment:
This math seems a bit confusing; should `sum_of_squares_` be literally the sum of the squares of `value`? Then we do the heavy lifting in `variance()` like in https://stackoverflow.com/a/1175084.

Reply:
I (tried to) implement knuths way of computing the running stats here, which should be numerically stable and reasonably accurate. The output looks good in tests; Maybe defer this to the track-for-future list for revisiting later for now? Maybe it would be nice to add test designed for catching numerical stability issues as well..

Reply:
ps - I switched everything to the population standard dev, which slightly simplifies this, and allows us to use a call from hdrhistogram_c, dropping some more complex code in the `HdrStatistic` implementation.

Reply:
Added a test for catastrophic cancellation, plus an implementation of the algorithm you pointed out. Added notes in the comments, see: https://github.com/envoyproxy/envoy-perf/pull/48/files#diff-653bc7d320cfd8bc55e04f79b81e5f0eR93
(side-note: the values used in the test for this proves variance will return negative values with the newly added SimpleStats implementation, which I think is an effect that  may have caused the 'negative sigma' errors when we took a look at fortio a while back).

Reply:
Yeah, it would be good to avoid the negative sigma possibility. It's fine using this algorithm, just providing good references to where it comes from would help the reader. Also, I think `sum_of_squares_` is not quite the right name, it's an aggregation of square distances, but there's a bit more going on than just sum of squares.